pyccl/core.py

```diff
@@ -800,33 +802,33 @@ def compute_linear_power(self):
         elif trf == 'boltzmann_isitgr':
             rescale_mg = False
             pk = get_isitgr_pk_lin(self)
-        elif trf == 'boltzmann_camb':
-            pk_nl_from_camb = False
-            if self._config_init_kwargs['matter_power_spectrum'] == "camb":
-                pk_nl_from_camb = True
-            pk = get_camb_pk_lin(self, nonlin=pk_nl_from_camb)
-            if pk_nl_from_camb:
-                pk, pk_nl = pk
-                self._pk_nl['delta_matter:delta_matter'] = pk_nl
-                self._has_pk_nl = True
-                rescale_mg = False
-                rescale_s8 = False
-                if abs(self["mu_0"]) > 1e-14:
-                    warnings.warn("You want to compute the non-linear power "
-                                  "spectrum using CAMB. This cannot be "
-                                  "consistently done with mu_0 > 0.",
-                                  category=CCLWarning)
-                if np.isfinite(self["sigma8"]) \
-                        and not np.isfinite(self["A_s"]):
-                    raise CCLError("You want to compute the non-linear "
-                                   "power spectrum using CAMB and specified "
-                                   "sigma8 but the non-linear power spectrum "
-                                   "cannot be consistenty rescaled.")
         elif trf in ['bbks', 'eisenstein_hu', 'eisenstein_hu_nowiggles']:
             rescale_s8 = False
             rescale_mg = False
             pk = Pk2D.from_model(self, model=trf)
 
+        # Compute the CAMB nonlin power spectrum if needed,
+        # to avoid repeating the code in `compute_nonlin_power`.
+        # Because CAMB power spectra come in pairs with pkl always computed,
+        # we set the nonlin power spectrum first, but keep the linear via a
+        # status variable to use it later if the transfer function is CAMB too.
+        pkl = None
+        if self._config_init_kwargs["matter_power_spectrum"] == "camb":
+            if abs(self["mu_0"]) > 1e-14:
+                warnings.warn("CAMB doesn't compute non-linear power spectra "
+                              "consistently with mu_0 > 0.", CCLWarning)
+            if np.isnan(self["A_s"]):
```

Comment:
Is this equivalent to `not np.isfinite`?

Reply:
It is not. `not np.isfinite(np.nan)` is `True`. `np.isnan(np.nan)` is `True`.
But `not np.isfinite(np.inf)` is `True` and `np.isnan(np.inf)` is `False`.
If not declared, `A_s` is `nan`, so we should be explicitly checking for nans.

Reply:
Since we set these to NAN at the C level, are we sure that the C NAN is the same as `np.nan` for all architectures or python/numpy versions? I perused a bit but couldn't confirm it.

Reply:
That's a good point, I have no idea. I think this is handled by SWIG. Would SWIG behave differently in different architectures?

Reply:
The main question is, if not `np.nan` then what? What is it replaced by? I can't think of anything that would fit in here.

Reply:
I don't know... I changed it back to `not np.isfinite` just to be on the safe side. Thanks for pointing that out.